src/LiveSplit.Delta/UI/Components/DeltaComponent.cs

```diff
@@ -156,6 +148,19 @@ public void Update(IInvalidator invalidator, LiveSplitState state, float width,
                 InternalComponent.TimeValue = null;
             }
 
+            var text = comparisonName;
+            if (Settings.OverrideText)
+            {
+                text = Settings.DifferentialText && InternalComponent.TimeValue < TimeSpan.Zero ? Settings.TextAhead : Settings.TextBehind;
+            }
+            if (InternalComponent.InformationName != text)
+            {
+                InternalComponent.AlternateNameText.Clear();
+                InternalComponent.AlternateNameText.Add(CompositeComparisons.GetShortComparisonName(comparison));
+            }
+            InternalComponent.LongestString = text;
```

Comment:
This needs to be the longest possible string out of all the values it can switch between, otherwise the horizontal layout will resize when the text changes.

Reply:
Specifically, I think it needs to be the longer of `Settings.CustomText` and `Settings.CustomTextAhead` when there's an override

Reply:
I can write it like that, though the old behaviour (when set to Current Comparison) was to update `LongestString` to the (current) long comparison name at each update, not the longest it can be without changing the settings.
If this variable is meant to be changed while running (previously it was at each current comparison change), I don't see the point of telling the renderer to make space for something longer than we need it to display.
If it's not meant to be changed regularly, would it be better to set it to the longest comparison name when set to current comparison ?
Or is it a kind of "middle ground", meaning changing comparison is meant to be relatively rare and at that point changing layout spacing isn't a big deal, while being ahead/behind changes often enough to warrant always making enough space for it ?

Reply:
The issue with your current code is that the text changes between ahead vs. behind, but `LongestString` doesn't account for this. That's the only case that's problematic in my opinion.

Changing the comparison or the settings isn't something we need to account for here